clients/fides-js/src/fides.ts

```diff
@@ -86,8 +90,23 @@ let _Fides: Fides;
 const updateCookie = async (
   oldCookie: FidesCookie,
   experience: PrivacyExperience,
-  debug?: boolean
+  debug?: boolean,
+  isExperienceClientSideFetched?: boolean
 ): Promise<{ cookie: FidesCookie; experience: PrivacyExperience }> => {
+  if (!isExperienceClientSideFetched) {
+    // If it's not client side fetched, we don't update anything since the cookie has already
+    // been updated earlier.
+    return { cookie: oldCookie, experience };
+  }
```

Comment:
I suspect this might be messing with our tests, which provide server side data with preferences, and we rely on those preferences for the tests... tests might be the only place where our server side data has preferences which we want to respect?

Reply:
yes I actually removed this bit, adding `isExperienceClientSideFetched` to the condition below it, so:

```
if (isExperienceClientSideFetched && oldCookie.consent) {
    // if we already have preferences on the cookie, update client-side experience with those preferences
    const updatedExperience = updateExperienceFromCookieConsent({
      experience,
      cookie: oldCookie,
      debug,
    });
    return { cookie: oldCookie, experience: updatedExperience };
  }
```